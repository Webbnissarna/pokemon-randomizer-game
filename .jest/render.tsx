import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
