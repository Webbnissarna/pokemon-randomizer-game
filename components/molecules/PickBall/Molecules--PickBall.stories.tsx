import { Flex } from "theme-ui";
import PickBall from ".";
import { makeLangGetter } from "../../../utils/GamePropSystem/LangGamePropSystem";
import { mockPokemon } from "../../../utils/testHelper";

interface ControlProps {
  showPokemon: boolean;
  showStar: boolean;
}

export function Default({ showPokemon, showStar }: ControlProps) {
  return (
    <Flex>
      <PickBall
        pokemon={mockPokemon}
        gamePropGetter={makeLangGetter("ja")}
        showPokemon={showPokemon}
        showStar={showStar}
        onClick={() => console.log("Click!")}
      />
    </Flex>
  );
}

Default.args = {
  showPokemon: false,
  showStar: false,
} as ControlProps;
