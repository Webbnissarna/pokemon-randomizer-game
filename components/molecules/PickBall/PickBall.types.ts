import { MouseEvent as ReactMouseEvent } from "react";
import { GamePropGetter } from "../../../utils/GamePropSystem/GamePropSystem.types";
import { GeneratedPokemon } from "../../../utils/Generator/generator.types";

export type ClickEvent = ReactMouseEvent<HTMLButtonElement, MouseEvent>;
export interface PickBallProps {
  pokemon: GeneratedPokemon;
  gamePropGetter: GamePropGetter<GeneratedPokemon>;
  showPokemon: boolean;
  showStar: boolean;
  onClick: (event: ClickEvent) => void;
}
