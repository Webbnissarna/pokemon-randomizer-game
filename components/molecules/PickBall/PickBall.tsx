import { Box, Flex, Image, Text } from "theme-ui";
import { toImgBase64PNG } from "../../../utils/Misc";
import Ball from "../../atoms/Ball";
import Star from "../../atoms/Star";
import { PickBallProps } from "./PickBall.types";

export default function PickBall({
  pokemon,
  gamePropGetter,
  showPokemon,
  showStar,
  onClick,
}: PickBallProps): JSX.Element {
  const gameProp = gamePropGetter(pokemon);

  return (
    <Flex
      sx={{
        flexDirection: "column",
        alignItems: "center",
        width: "fit-content",
        gap: "sm",
      }}
    >
      <Box
        sx={{
          position: "relative",
        }}
      >
        <Box
          sx={{
            position: "absolute",
            top: "calc(50% - 90px)",
            left: "calc(50% - 90px)",
          }}
        >
          <Star size={180} visible={showStar} />
        </Box>
        <Box
          sx={{
            position: "relative",
          }}
        >
          <Box
            sx={{
              ...(showPokemon && { opacity: 0 }),
            }}
          >
            <Ball size="lg" onClick={onClick} />
          </Box>
          {showPokemon && (
            <Box
              sx={{
                position: "absolute",
                top: 0,
                left: 0,
                right: 0,
                bottom: 0,
                padding: "md",
                pointerEvents: "none",
              }}
            >
              <Image
                src={toImgBase64PNG(pokemon.info.imageData)}
                alt={pokemon.info.name.en}
              />
            </Box>
          )}
        </Box>
      </Box>
      <Text
        sx={{
          fontSize: "lg",
          zIndex: 1,
        }}
      >
        {gameProp.mainProp}
      </Text>
      {gameProp.optionalProp && (
        <Text
          sx={{
            fontSize: "lg",
          }}
        >
          {gameProp.optionalProp}
        </Text>
      )}
    </Flex>
  );
}
