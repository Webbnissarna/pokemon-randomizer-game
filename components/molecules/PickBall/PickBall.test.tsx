import PickBall from ".";
import { render } from "../../../.jest/render";
import {
  mockPokemon,
  mockPropGetter,
  MOCK_PROP,
} from "../../../utils/testHelper";

describe("PickBall", () => {
  it("renders", () => {
    const { asFragment } = render(
      <PickBall
        gamePropGetter={mockPropGetter}
        pokemon={mockPokemon}
        onClick={() => undefined}
        showPokemon={false}
        showStar={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows GameProp properties", () => {
    const { getByText } = render(
      <PickBall
        gamePropGetter={mockPropGetter}
        pokemon={mockPokemon}
        onClick={() => undefined}
        showPokemon={false}
        showStar={false}
      />
    );

    expect(getByText(MOCK_PROP.mainProp)).toBeTruthy();
    expect(getByText(MOCK_PROP.optionalProp)).toBeTruthy();
  });
});
