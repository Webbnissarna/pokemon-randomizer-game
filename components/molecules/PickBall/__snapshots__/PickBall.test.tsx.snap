// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickBall renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-1evffay-Box"
  >
    <div
      class="css-gvtkf1-Box"
    >
      <div
        class="css-coasxz-Box"
      >
        <div
          class="css-17zq141-Box"
          title="star"
        >
          <div
            class="css-1envnbc-Box"
          >
            <svg
              height="95.10545014740343%"
              style="will-change: transform; shape-rendering: crispEdges;"
              version="1.1"
              viewBox="0 0 26.458 25.163"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m-42.235 84.622-11.167-34.625-36.381 0.07952 29.48-21.32-11.318-34.576 29.386 21.449 29.386-21.449-11.318 34.576 29.48 21.32-36.381-0.079517z"
                fill="var(--theme-ui-colors-star)"
                stop-color="#000000"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2.5159"
                transform="matrix(-.27823 0 0 -.27823 1.4781 23.544)"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="css-gvtkf1-Box"
      >
        <div
          class="css-1peva9n-Box"
        >
          <button
            class="css-1s9y82o-Box"
          >
            <svg
              height="100"
              style="filter: drop-shadow(rgba(0, 0, 0, 0.25) 0px 2px 2px);"
              version="1.1"
              viewBox="0 0 26.458 26.458"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                stroke-linecap="round"
                stroke-linejoin="round"
                transform="translate(-91.771 -141.89)"
              >
                <path
                  d="m105 141.89a13.229 13.229 0 0 0-13.229 13.229h26.458a13.229 13.229 0 0 0-13.229-13.229z"
                  fill="var(--theme-ui-colors-ball-active-top)"
                  stop-color="#000000"
                  stroke-width=".7"
                />
                <path
                  d="m105 168.34a13.229 13.229 0 0 0 13.229-13.229h-26.458a13.229 13.229 0 0 0 13.229 13.229z"
                  fill="var(--theme-ui-colors-ball-active-bottom)"
                  stop-color="#000000"
                  stroke-width=".7"
                />
                <g
                  fill="var(--theme-ui-colors-ball-active-middle)"
                >
                  <rect
                    height="1.8521"
                    stop-color="#000000"
                    stroke-width=".7"
                    width="26.458"
                    x="91.771"
                    y="153.26"
                  />
                  <rect
                    height="1.8521"
                    stop-color="#000000"
                    stroke-width=".7"
                    width="26.458"
                    x="91.771"
                    y="155.11"
                  />
                  <circle
                    cx="105"
                    cy="155.11"
                    r="4.6302"
                    stop-color="#000000"
                    stroke-width=".7"
                  />
                </g>
                <circle
                  cx="105"
                  cy="155.11"
                  fill="var(--theme-ui-colors-ball-active-bottom)"
                  r="2.6458"
                  stop-color="#000000"
                  stroke-width=".7"
                />
                <path
                  d="m50 0a50 50 0 0 0-50 50 50 50 0 0 0 20.811 40.557 50 50 0 0 1-11.561-31.932 50 50 0 0 1 50-50 50 50 0 0 1 29.203 9.418 50 50 0 0 0-38.453-18.043z"
                  fill-opacity=".25"
                  stop-color="#000000"
                  stroke-width="2.6457"
                  transform="matrix(.26458 0 0 .26458 91.771 141.89)"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <span
      class="css-1tgkd6o-Box"
    >
      mainMock
    </span>
    <span
      class="css-2vsnqm-Box"
    >
      optionalMock
    </span>
  </div>
</DocumentFragment>
`;
